src/network/routing/lokinet_router.cpp

```diff
@@ -0,0 +1,648 @@
+#include "session/network/routing/lokinet_router.hpp"
+
+#include <fmt/ranges.h>
+#include <fmt/std.h>
+#include <oxenc/base64.h>
+
+#include <llarp/contact/router_id.hpp>
+#include <lokinet.hpp>
+#include <oxen/log.hpp>
+#include <oxen/log/format.hpp>
+
+#include "session/network/network_opt.hpp"
+#include "session/onionreq/builder.hpp"
+#include "session/onionreq/response_parser.hpp"
+
+using namespace oxen;
+using namespace session;
+using namespace session::network;
+using namespace std::literals;
+using namespace oxen::log::literals;
+
+namespace session::network {
+
+namespace {
+    auto cat = oxen::log::Cat("network");
+
+    static constexpr std::string_view PROXIED_REQUESTS_KEY{"proxied_requests"};
+
+    std::string pending_request_key(const network_destination& dest) {
+        std::optional<std::string> key;
+
+        std::visit(
+                [&key](auto&& arg) {
+                    using T = std::decay_t<decltype(arg)>;
+
+                    if constexpr (std::is_same_v<T, oxen::quic::RemoteAddress>) {
+                        key = oxenc::to_hex(arg.view_remote_key());
+                    } else if constexpr (std::is_same_v<T, service_node>) {
+                        key = oxenc::to_hex(arg.view_remote_key());
+                    } else if constexpr (std::is_same_v<T, ServerDestination>) {
+                        key = PROXIED_REQUESTS_KEY;
+                    }
```

Comment:
Currently if the code runs off the bottom of this if / else if / else if then `key` remains nullopt, and later on we throw a runtime_error.

All of that can be detected at compile time, however, by changing this final `else if` into an `else` with a static assert:
```suggestion
                    } else {
                        static_assert(std::is_same_v<T, ServerDestination>);
                        key = PROXIED_REQUESTS_KEY;
                    }
```
That way, if something changes that adds another type to the variant, this code simply won't compile rather than causing runtime errors.